web/src/filter.ts

```diff
@@ -343,6 +405,9 @@ export class Filter {
                 // Looks like an operator.
                 negated = false;
                 operator = parts.shift();
+                if (operator === undefined) {
+                    continue;
```

Comment:
This is creating codecov issues. Is it reasonable to expect it to always get something from `parts`? It seems like it has thus far without throwing errors

Reply:
It looks like this will always be defined since `.split` returns a nonempty array and we’ve checked that the length isn’t 1. This isn’t performance-sensitive enough to warrant an unchecked `!`, so `assert(operator !== undefined)` is a fine short-term fix.

A better way to rewrite this would be using `str.matchAll` rather than `str.match` to get out the match objects with individual capture groups, so we don’t need to write a separate nonsense parser with `.split`.

Reply:
I can do this in a followup PR :)